src/ovp_pipeline/mcp_server.py

```
@@ -0,0 +1,429 @@
+            return {"tools": self.list_tools()}
+        if method == "tools/call":
+            name = params.get("name")
+            arguments = params.get("arguments") or {}
+            if not isinstance(name, str) or not name:
+                raise _InvalidParams("tools/call requires a string 'name'")
+            if not isinstance(arguments, dict):
+                raise _InvalidParams("tools/call 'arguments' must be an object")
+            try:
+                result = self.call_tool(name, arguments)
+            except KeyError:
+                raise _InvalidParams(f"Unknown tool: {name}")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)